include/seqan3/argument_parser/validators.hpp

```diff
@@ -62,12 +62,12 @@ namespace seqan3
  */
 //!\cond
 template <typename validator_type>
-concept validator_concept = std::Invocable<validator_type, typename validator_type::value_type> &&
-                                 requires(validator_type validator,
-                                          typename validator_type::value_type value)
+concept validator_concept = requires(validator_type validator,
+                                     typename std::remove_reference_t<validator_type>::value_type value)
```

Comment:
you requesting a type before it is checked in the concept. I am not sure, what happens when the type does not support a value_type member? Will this result in a compiler error or will it evaluate to false?

Reply:
I couldn't find a negative test case either.

Reply:
I think this type must be given as a second template parameter, like the invocable concepts from the STL.

Reply:
> you requesting a type before it is checked in the concept. I am not sure, what happens when the type does not support a value_type member? Will this result in a compiler error or will it evaluate to false?

tried and tested: This will not cause a compiler error but just result in the concept being not satisfied.

> I couldn't find a negative test case either.

I'll add one

> I think this type must be given as a second template parameter, like the invocable concepts from the STL.
 
no. See first answer.